homeassistant/components/switchbot/__init__.py

```diff
@@ -1 +1,80 @@
-"""The switchbot component."""
+"""Support for Switchbot devices."""
+from asyncio import Lock
+
+import switchbot  # pylint: disable=import-error
+
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    CONF_RETRY_COUNT,
+    CONF_RETRY_TIMEOUT,
+    CONF_SCAN_TIMEOUT,
+    CONF_TIME_BETWEEN_UPDATE_COMMAND,
+    DATA_COORDINATOR,
+    DEFAULT_RETRY_COUNT,
+    DEFAULT_RETRY_TIMEOUT,
+    DEFAULT_SCAN_TIMEOUT,
+    DEFAULT_TIME_BETWEEN_UPDATE_COMMAND,
+    DOMAIN,
+)
+from .coordinator import SwitchbotDataUpdateCoordinator
+
+PLATFORMS = ["switch"]
+
+
+async def async_setup_entry(hass, entry) -> bool:
+    """Set up Switchbot from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if not entry.options:
+        options = {
+            CONF_TIME_BETWEEN_UPDATE_COMMAND: DEFAULT_TIME_BETWEEN_UPDATE_COMMAND,
+            CONF_RETRY_COUNT: DEFAULT_RETRY_COUNT,
+            CONF_RETRY_TIMEOUT: DEFAULT_RETRY_TIMEOUT,
+            CONF_SCAN_TIMEOUT: DEFAULT_SCAN_TIMEOUT,
+        }
+
+        hass.config_entries.async_update_entry(entry, options=options)
+
+    if not hass.data[DOMAIN].get(DATA_COORDINATOR):
+        switchbot.DEFAULT_RETRY_TIMEOUT = entry.options[CONF_RETRY_TIMEOUT]
+
+        # BTLE has issues with multiple connections,
+        # so we use a lock to ensure that only one API request is reaching it at a time:
+        api_lock = Lock()
```

Comment:
Out of scope for this pr, but it would probably be useful to have this as a global lock in Home Assistant, so all integrations with bt could use the same lock

Reply:
Maybe a central broker to obtain btle access.